src/hooks/useForm/useTraditionalForm.tsx

```diff
@@ -79,6 +77,27 @@ const useTraditionalForm = ({
     updateQuestionVisibility()
   }, [updateQuestionVisibility])
 
+  // Mark all initial values as dirty
+  useEffect(() => {
+    if (autosaveAnswers && !isEmpty(initialValues)) {
+      formMethods.reset(getInitialValues(questions))
+      Object.keys(initialValues).forEach((key) => {
+        formMethods.setValue(key, initialValues[key], {
+          shouldDirty: true,
+          shouldTouch: true,
+        })
+      })
+      formMethods.trigger().then(() => {
+        // Ensure validation is completed before updating visibility
+        const allValues = formMethods.getValues()
+        if (onAnswersChange) {
+          onAnswersChange(JSON.stringify(allValues) ?? '{}')
+        }
+        updateQuestionVisibility()
+      })
+    }
+  }, [])
+
```

Comment:
Code looks quite similar, does it make sense to have a helper that can be shared in both hooks? It is also inside useEffect which looks like a good candidate ;-)